From: David Woodhouse <dwmw@amazon.co.uk>
Date: Fri, 15 Dec 2017 05:43:29 -0800
Subject: x86: Add STIBP feature enumeration
Patch-mainline: submitted on 2018/1/9
References: bnc#1068032 CVE-2017-5715

Enumerate single thread indirect branch predictors (STIBP) feature. It
provides means to prevent indirect branch predictions from being
controlled by sibling HW thread.

Signed-off-by: David Woodhouse <dwmw@amazon.co.uk>
Signed-off-by: Tim Chen <tim.c.chen@linux.intel.com>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 arch/x86/include/asm/cpufeatures.h | 1 +
 arch/x86/kernel/cpu/scattered.c    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/x86/include/asm/cpufeatures.h b/arch/x86/include/asm/cpufeatures.h
index 8cb58ca5d298..f00024529863 100644
--- a/arch/x86/include/asm/cpufeatures.h
+++ b/arch/x86/include/asm/cpufeatures.h
@@ -210,6 +210,7 @@
 
 #define X86_FEATURE_MBA			( 7*32+18) /* Memory Bandwidth Allocation */
 #define X86_FEATURE_SPEC_CTRL		( 7*32+19) /* Control Speculation Control */
+#define X86_FEATURE_STIPB		( 7*32+20) /* Single Thread Indirect Branch Predictors */
 #define X86_FEATURE_IA32_ARCH_CAPS	( 7*32+21) /* Control Speculation Control */
 #define X86_FEATURE_IBRS_ATT		( 7*32+22) /* IBRS all the time */
 
diff --git a/arch/x86/kernel/cpu/scattered.c b/arch/x86/kernel/cpu/scattered.c
index b2123c67bf85..e44c1ffe4ece 100644
--- a/arch/x86/kernel/cpu/scattered.c
+++ b/arch/x86/kernel/cpu/scattered.c
@@ -25,6 +25,7 @@ static const struct cpuid_bit cpuid_bits[] = {
 	{ X86_FEATURE_AVX512_4VNNIW,    CPUID_EDX,  2, 0x00000007, 0 },
 	{ X86_FEATURE_AVX512_4FMAPS,    CPUID_EDX,  3, 0x00000007, 0 },
 	{ X86_FEATURE_SPEC_CTRL,	CPUID_EDX, 26, 0x00000007, 0 },
+	{ X86_FEATURE_STIPB,		CPUID_EDX, 27, 0x00000007, 0 },
 	{ X86_FEATURE_IA32_ARCH_CAPS,	CPUID_EDX, 29, 0x00000007, 0 },
 	{ X86_FEATURE_CAT_L3,		CPUID_EBX,  1, 0x00000010, 0 },
 	{ X86_FEATURE_CAT_L2,		CPUID_EBX,  2, 0x00000010, 0 },
-- 
2.15.1

